panel/tests/widgets/test_chat.py

```diff
@@ -0,0 +1,1125 @@
+        entry = ChatEntry(value=component())
+        assert not entry.chat_copy_icon.visible
+        assert not entry.chat_copy_icon.value
+
+class TestChatFeed:
+    @pytest.fixture
+    def chat_feed(self):
+        return ChatFeed()
+
+    def test_hide_header(self, chat_feed):
+        assert chat_feed.header is None
+
+        chat_feed.header = "# Header"
+        assert not chat_feed._composite.hide_header
+
+        chat_feed.header = None
+        assert chat_feed._composite.hide_header
+
+        chat_feed.header = ""
+        assert chat_feed._composite.hide_header
+
+    def test_send(self, chat_feed):
+        entry = chat_feed.send("Message")
+        assert len(chat_feed.value) == 1
+        assert chat_feed.value[0] is entry
+        assert chat_feed.value[0].value == "Message"
+
+    def test_link_chat_log_objects(self, chat_feed):
+        chat_feed.send("Message")
+        assert chat_feed._chat_log.objects[0] is chat_feed.value[0]
+
+    def test_send_with_user_avatar(self, chat_feed):
+        user = "Bob"
+        avatar = "👨"
+        entry = chat_feed.send("Message", user=user, avatar=avatar)
+        assert entry.user == user
+        assert entry.avatar == avatar
+
+    def test_send_dict(self, chat_feed):
+        entry = chat_feed.send({"value": "Message", "user": "Bob", "avatar": "👨"})
+        assert len(chat_feed.value) == 1
+        assert chat_feed.value[0] is entry
+        assert chat_feed.value[0].value == "Message"
+        assert chat_feed.value[0].user == "Bob"
+        assert chat_feed.value[0].avatar == "👨"
+
+    def test_send_dict_minimum(self, chat_feed):
+        entry = chat_feed.send({"value": "Message"})
+        assert len(chat_feed.value) == 1
+        assert chat_feed.value[0] is entry
+        assert chat_feed.value[0].value == "Message"
+
+    def test_send_dict_without_value(self, chat_feed):
+        with pytest.raises(ValueError, match="it must contain a 'value' key"):
+            chat_feed.send({"user": "Bob", "avatar": "👨"})
+
+    def test_send_dict_with_user_avatar_override(self, chat_feed):
+        user = "August"
+        avatar = "👩"
+        entry = chat_feed.send(
+            {"value": "Message", "user": "Bob", "avatar": "👨"},
+            user=user,
+            avatar=avatar,
+        )
+        assert len(chat_feed.value) == 1
+        assert chat_feed.value[0] is entry
+        assert chat_feed.value[0].value == "Message"
+        assert chat_feed.value[0].user == user
+        assert chat_feed.value[0].avatar == avatar
+
+    def test_send_entry(self, chat_feed):
+        entry = ChatEntry(value="Message", user="Bob", avatar="👨")
+        chat_feed.send(entry)
+        assert len(chat_feed.value) == 1
+        assert chat_feed.value[0] is entry
+        assert chat_feed.value[0].value == "Message"
+        assert chat_feed.value[0].user == "Bob"
+        assert chat_feed.value[0].avatar == "👨"
+
+    def test_send_entry_with_user_avatar_override(self, chat_feed):
```

Comment:
I don't think we should allow this.

Reply:
I've got a relatively large refactor of some of the internals and it explicitly disallows this.